### ISSUE TITLE
Add hatch equivalent of a MANIFEST to ignore non-code file for sdist

### ISSUE THREAD

**Issue by matthewfeickert** [peterridolfi/Pyhf-to-Combine-converter#21]
I'm putting this here as a note for later, but after PR #20 was added I noticed that for the sdist there are lots of non-code files that should be ignored

```console
$ python -m tarfile --list dist/pyhf_combine_converter-*.tar.gz
pyhf_combine_converter-0.0.3.dev9+g857007f/.github/workflows/publish-package.yml 
pyhf_combine_converter-0.0.3.dev9+g857007f/docs/Main_differences.md 
pyhf_combine_converter-0.0.3.dev9+g857007f/docs/Twice_translated_card/verification_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/docs/Twice_translated_card/verification_datacard_original.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/src/pyhf_combine_converter/__init__.py 
pyhf_combine_converter-0.0.3.dev9+g857007f/src/pyhf_combine_converter/pyhf_convert_to_datacard.py 
pyhf_combine_converter-0.0.3.dev9+g857007f/src/pyhf_combine_converter/pyhf_converted_from_datacard.py 
pyhf_combine_converter-0.0.3.dev9+g857007f/src/pyhf_combine_converter/version.py 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/combine_logger.out 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/pull_plot.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/to_pyhf_test.py 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/NLLplot.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/NLLplot_shape.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/exp_data_counting.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/exp_data_counting_varyparams_combined.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/exp_data_shape.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/exp_data_shape_varyparams.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/fitDiagnostics_counting.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/fitDiagnostics_shape.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/fitDiagnostics_shape_vary_params.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/Validation/higgsCombineTest.MultiDimFit.mH120.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/big_model/bottom-squarks.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/big_model/converted_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/big_model/shapes.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/big_model/workspace_Comb.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_counting_analysis/counting_fit_combine.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_counting_analysis/covariance_fit_b_counting_combine.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_counting_analysis/covariance_fit_s_counting_combine.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_counting_analysis/plots.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/bin1_CMS_th1x_fit_b.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/bin1_CMS_th1x_fit_s.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/bin1_CMS_th1x_prefit.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/covariance_fit_b_shape_combine.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/covariance_fit_s_shape_combine.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/combine_shape_analysis/shapes_fit_combine.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/bin1_prefit.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/correlation_matrix_counting.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/counting_pulls_pyhf.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/e_mu_prefit.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/e_tau_prefit.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_counting_analysis/mu_tau_prefit.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_shape_analysis/bin1_prefit.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_shape_analysis/correlation_matrix_shape.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_shape_analysis/shape_pulls_pyhf.pdf 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/figures/pyhf_shape_analysis/shapes_brazil_band.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/NLLplot_AGREEMENT.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/higgsCombineTest.MultiDimFit.mH120_1.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/higgsCombineTest.MultiDimFit.mH120_2.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/higgsCombineTest.MultiDimFit.mH120_multi.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/shapes.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/single_bin_1_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/single_bin_1_test.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/single_bin_2_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/single_bin_2_test.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/two_bin_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/multi_vs_single_bin/two_bin_test.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplotVERIFICATION.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplot_FULL_VERIFICATION_INTEGERS_ONLY.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplot_SHAPES.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplot_floats.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplot_profiling.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/NLLplot_staterror.png 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/combine_logger.out 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/converted_datacard.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/converted_workspace_floats.json 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/higgsCombineTest.MultiDimFit.mH120.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/higgsCombineTest.MultiDimFit.mH120_fitscan.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/shapes-test.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/test/verification_model/shapes.root 
pyhf_combine_converter-0.0.3.dev9+g857007f/.gitignore 
pyhf_combine_converter-0.0.3.dev9+g857007f/LICENSE.txt 
pyhf_combine_converter-0.0.3.dev9+g857007f/README.md 
pyhf_combine_converter-0.0.3.dev9+g857007f/pyproject.toml 
pyhf_combine_converter-0.0.3.dev9+g857007f/PKG-INFO 
```

we only want the files under `src` and then `LICENSE.txt`, `README.MD`, and `pyproject.toml`.

cc @agoose77 though Angus don't spend time on this right away unless you know immediately how to do this as I haven't put time into it yet.

**Comment by agoose77** [comment]
Usually one wants to include much of the VCS in the sdist so that it represents a snapshot of the repository (minus the git metadata). But, if you only want to include the Python package files and metadata, you can replace `[tool.hatch]` with 
```toml
[tool.hatch]
version.source = "vcs"

[tool.hatch.build]
only-packages = true

[tool.hatch.build.hooks.vcs]
version-file = "src/pyhf_combine_converter/version.py"
```
You can scope this to a particular build target, e.g. sdist with:
```toml
[tool.hatch]
version.source = "vcs"

[tool.hatch.build.sdist]
only-packages = true

[tool.hatch.build.hooks.vcs]
version-file = "src/pyhf_combine_converter/version.py"
```

If you only want these files removed from the wheel, then you're already done with
```toml
[tool.hatch]
version.source = "vcs"

[tool.hatch.build.hooks.vcs]
version-file = "src/pyhf_combine_converter/version.py"
```
because Hatch defaults to include only the package for wheel targets if you don't specify any include options: https://hatch.pypa.io/latest/plugins/builder/wheel/#default-file-selection

**Comment by matthewfeickert** [comment]
Another `hatch-vcs` related question: https://github.com/ofek/hatch-vcs/discussions/12

**Comment by matthewfeickert** [comment]
I should also check out [`mario-mapyde`'s `pyproject.toml`](https://github.com/scipp-atlas/mario-mapyde/blob/66a27540ce166ede99f4ea85703ba5c4d7ed6d68/pyproject.toml) written by @kratsg.

**Comment by matthewfeickert** [comment]
Resolved by PR #35
